### PR TITLE
Minor code improvements and addition test set

### PR DESCRIPTION
**Pull request by Sim589**
Have improved the neural net code by adding the ability to save to github, and then load to allow the user to not have to retrain a network every time. The other notable improvement is to be able to carry where you left off in training (if it was already trained for 10 epochs, you can now train another 5 epochs no problem). I also added a test set of 20 actual tube images

### CODE REVIEW

**Comment by Sim589** [comment]
Sorry for overriding Chris, just wanted to get this done fast
